System/Process.hs

```diff
@@ -870,6 +869,22 @@ It will therefore behave more portably between operating systems than 'system'.
 The return codes and possible failures are the same as for 'system'.
 -}
 rawSystem :: String -> [String] -> IO ExitCode
-rawSystem cmd args = do
-  (_,_,_,p) <- createProcess_ "rawSystem" (proc cmd args) { delegate_ctlc = True }
+rawSystem cmd args = executeAndWait "rawSystem" (proc cmd args) { delegate_ctlc = True }
+
+-- ---------------------------------------------------------------------------
+-- executeAndWait
+
+-- | Create a new process and wait for it's termination.
```

Comment:
@snoyberg By default, yes. However `CreateProcess` has a flag `CREATE_BREAKAWAY_FROM_JOB` that allows a child process to break away from any parent job. So you can get the same behaviour as on linux by `B` using this flag when it spawns `C`. Though I agree, it is subtle.. (There's also a flag to prevent any children from breaking away, so it goes both ways :))

@simonmar I'll update the comment. 
So there are two alternate approaches that could be taken here:
1) always use a job. In which case, if a user hasn't been deconstructing `ProcessHandle` then it should work transparently.
2) Combine the new function with the old one instead of having two calls now, and make the behaviour implicit on flag. But this would not be backwards compatible, since of course there's a new parameter.

I went with separate functions to avoid always using a job and breaking backwards compatibility. 

If we go for 1) then, according to some very statistically unsound testing, the added overhead is about `0.0021 ms`.